src/com/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -883,6 +884,108 @@
     (sql/do-commands
      "DROP INDEX fact_values_string_trgm")))
 
+(defn- lift-fact-paths-into-facts
+  "Pairs paths and values directly in facts, i.e. change facts from (id
+  value) to (id path value)."
+  []
+  (sql/do-commands
+   (if (scf-utils/postgres?)
+     "SET CONSTRAINTS ALL DEFERRED" ;; In case it allows additional optimization?
```

Comment:
@wkalt as per other discussion, I'll be removing deferrals when possible in the next revision.